src/main/java/mchorse/metamorph/api/morphs/EntityMorph.java

```diff
@@ -373,6 +376,42 @@ public void render(EntityLivingBase entity, double x, double y, double z, float
             renderEntity = null;
         }
     }
+    
+    /* Entity morph settings */
+
+    /**
+     * These are settings that are defined from the morph entity.
+     * They have lower priority than activeSettings.
+     */
+    protected MorphSettings entitySettings = null;
+    
+    protected void setEntitySettings(MorphSettings entitySettings) {
+        this.entitySettings = entitySettings;
+        this.needSettingsUpdate = true;
+    }
+    
+    @Override
+    public void initializeSettings()
```

Comment:
`EntityMorph.initializeSettings` is now where the most important settings initialization happens. There is also a simpler base implementation in `AbstractMorph.initializeSettings`